### PR TITLE
Get realtime database saving on the edit and add user pages

### PR DESCRIPTION
**Pull request by mrchrisadams**
This is PR adds the required code  for listing tags on the add user and edit profile screens in constellate.

Where most of the state is manipulated through the store with VueX, the lists of tags where generated by using the VueFire extension, and the single file components needed to be updated to support the editing and creation of tags, using the newer syntax.

You can see more below, on the VueFire docs.
https://vuefire.vuejs.org/vuefire/upgrading-from-v1.html#renamed-import-for-the-rtdb

### CODE REVIEW

**Comment by mrchrisadams** [comment]
Ah, nuts.

This includes the changes to `config.js`, that really belong in another PR, as they're updating the server side changes, and admin scripts.

I'll take that out, so we can merge this in cleanly

**Comment by mrchrisadams** [comment]
Okay, this is added. @rmparr , can you look over check this makes sense?

The handling of tags is one area that is a bit tricky, and this reintroduces the missing functionality.

File: cl8-web/src/store.js [hunk 1]
```diff
@@ -386,17 +386,17 @@ export default new Vuex.Store({
     updateProfile: function(context, payload) {
       debug('sending update to Firebase', payload)
 
-      // not super happy about this - surely there's a toJSON() method?
+      // doing this round trip returns a JSON object we
+      // can save back to the realtime database more easily,
+      // and strips out properties we wouldn't want to save into it
       let newProfile = JSON.parse(JSON.stringify(payload))
```
Comment (rmparr) [inline cl8-web/src/store.js:392]: What is `newProfile` doing now, with `pushKey` now assigned to `payload['.key']`?

If it's getting set to something new below, can it just be assigned to undefined first?

Comment (mrchrisadams) [reply cl8-web/src/store.js:392]: newProfile is acting as the updated JSON object that we want to save to firebase, with all the properties stripped out of the object that are added by something like Vue, and Vuefire.

The pushkey is the key used to work out where to put the updated user object in the userlist. It used to be in the JSON object we would store in the userlist, but it looks like it's not longer added by Vue, so we don't need to change it.

I guess we could make this a `const` now, as it doesn't change, and we don't need to remove the key anymore.

Does that help? I'm happy to do a quick call,  as this bit wasn't that intuitive when I first started working with it.

Comment (rmparr) [reply cl8-web/src/store.js:392]: I'm good with that. It wasn't clear to me with how the key was being assigned previously, but that all makes sense.

**Pull request merged**